cupy/cuda/cub.pyx

```diff
@@ -124,12 +124,20 @@ def device_reduce(ndarray x, op, out=None, bint keepdims=False):
     y_ptr = <void *>y.data.ptr
     dtype_id = _get_dtype_id(x.dtype)
     s = <Stream_t>stream.get_current_stream_ptr()
-
+    x_size = <int>x.size
     ws_size = cub_device_reduce_get_workspace_size(x_ptr, y_ptr, x.size, s,
                                                    op, dtype_id)
     ws = memory.alloc(ws_size)
     ws_ptr = <void *>ws.ptr
-    cub_device_reduce(ws_ptr, ws_size, x_ptr, y_ptr, x.size, s, op, dtype_id)
+    with nogil:
+        cub_device_reduce(ws_ptr,
+                          ws_size,
+                          x_ptr,
+                          y_ptr,
+                          x_size,
+                          s,
+                          op,
+                          dtype_id)
```

Comment:
This line breaking is horrible. Any reason the style can't be preserved? This should be better:
```python
# I didn't do the character count, so this breaking point might be bad
with nogil:
    cub_device_reduce(ws_ptr, ws_size, x_ptr, y_ptr, x_size, s, 
                      op, dtype_id)  
```